nodeEditor/src/diagram/properties/genericNodePropertyComponent.tsx

```diff
@@ -39,7 +39,13 @@ export class GeneralPropertyTabComponent extends React.Component<IPropertyCompon
                     {
                         (!this.props.block.isInput || !(this.props.block as InputBlock).isAttribute) &&
                         <TextInputLineComponent globalState={this.props.globalState} label="Name" propertyName="name" target={this.props.block}
-                            onChange={() => this.props.globalState.onUpdateRequiredObservable.notifyObservers()} />
+                            onChange={ newName => 
+                            {if(!this.props.block.validateBlockName(newName)){ 
+                                this.props.globalState.onErrorMessageDialogRequiredObservable.notifyObservers(`"${newName}" is a reserved name, please choose another`); 

```

Comment:
I have a problem with that because this.props.block is already changed here
I would recommend adding a new feature to TextInputLineComponent named validator which is an optional function like (value: string) => boolean

You can then use that validator to block the update and display the message